modules/mysql/main.tf

```diff
@@ -0,0 +1,194 @@
+    }
+
+    maintenance_window {
+      day          = "${lookup(var.read_replica, "maintenance_window_day", local.default_maintenance_window_day)}"
+      hour         = "${lookup(var.read_replica, "maintenance_window_hour", local.default_maintenance_window_hour)}"
+      update_track = "${lookup(var.read_replica, "maintenance_window_update_track", local.default_maintenance_window_update_track)}"
+    }
+  }
+
+  depends_on = ["google_sql_database_instance.master"]
+
+  lifecycle {
+    ignore_changes = ["disk_size"]
+  }
+}
+
+resource "google_sql_database_instance" "failover-replica" {
+  count                 = "${length(var.failover_replica) == 0 ? 0 : 1}"
+  project               = "${var.project_id}"
+  name                  = "${var.name}-failover"
+  database_version      = "${var.database_version}"
+  region                = "${var.region}"
+  master_instance_name  = "${google_sql_database_instance.master.name}"
+  replica_configuration = ["${merge(var.replica_configuration, map("failover_target", true))}"]
+
+  settings {
+    tier                        = "${lookup(var.failover_replica, "tier", local.default_tier)}"
+    activation_policy           = "${lookup(var.failover_replica, "activation_policy", local.default_activation_policy)}"
+    authorized_gae_applications = ["${var.authorized_gae_applications}"]
+    ip_configuration            = ["${var.ip_configuration}"]
+
+    crash_safe_replication = "${lookup(var.failover_replica, "crash_safe_replication", local.default_crash_safe_replication)}"
+    disk_autoresize        = "${lookup(var.failover_replica, "disk_autoresize", local.default_disk_autoresize)}"
+    disk_size              = "${lookup(var.failover_replica, "disk_size", local.default_disk_size)}"
+    disk_type              = "${lookup(var.failover_replica, "disk_type", local.default_disk_type)}"
+    pricing_plan           = "${lookup(var.failover_replica, "pricing_plan", local.default_pricing_plan)}"
+    replication_type       = "${lookup(var.failover_replica, "replication_type", local.default_replication_type)}"
+    user_labels            = "${var.failover_replica_labels}"
+    database_flags         = ["${var.failover_replica_database_flags}"]
+
+    location_preference {
+      zone = "${var.region}-${lookup(var.failover_replica, "zone")}"
+    }
+
+    maintenance_window {
+      day          = "${lookup(var.failover_replica, "maintenance_window_day", local.default_maintenance_window_day)}"
+      hour         = "${lookup(var.failover_replica, "maintenance_window_hour", local.default_maintenance_window_hour)}"
+      update_track = "${lookup(var.failover_replica, "maintenance_window_update_track", local.default_maintenance_window_update_track)}"
+    }
+  }
+
+  depends_on = ["google_sql_database_instance.master"]
+
+  lifecycle {
+    ignore_changes = ["disk_size"]
+  }
+}
+
+resource "google_sql_database" "databases" {
```

Comment:
Since databases functionality isn't provided today, I think let's save this for a follow-on PR.

Reply:
Well, it should be supported by [Databases: insert](https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/databases/insert?hl=en).
In fact, multiple databases can be created by using this resource.

Reply:
That makes sense. To avoid big migration, I think we should also switch `google_sql_database` and `google_sql_user` back to `default`.

Reply:
Fixed: https://github.com/GoogleCloudPlatform/terraform-google-sql-db/pull/21/commits/de16a90bcde43968ab888850a72f4962fa40232e